web/onos-gui/src/app/onos-topo/device-detail/device-detail.component.html

```diff
@@ -73,52 +65,59 @@ <h2 class="editable clickable">{{detailsData?.getId()}}</h2>
             <hr>
         </div>
         <div class="middle">
+            <h2 class="changes-title">Attributes</h2>
+            <table>
+                <tbody>
+                    <tr *ngFor="let attribute of detailsData?.getAttributesMap()['map_'] | keyvalue; let i = index">
+                        <div *ngIf="i % 2 == 0" class="row">
+                            <td class="label" width="110">
+                                {{objToArr(detailsData?.getAttributesMap()['map_'])[i].key}} </td>
+                            <td class="value" width="110">
+                                {{objToArr(detailsData?.getAttributesMap()['map_'])[i].value}} </td>
+                            <td class="label" width="110"
+                                *ngIf="i + 1 < objToArr(detailsData?.getAttributesMap()['map_']).length">
+                                {{objToArr(detailsData?.getAttributesMap()['map_'])[i + 1].key}} </td>
+                            <td class="value" width="110"
+                                *ngIf="i + 1 < objToArr(detailsData?.getAttributesMap()['map_']).length">
+                                {{objToArr(detailsData?.getAttributesMap()['map_'])[i + 1].value}} </td>
+                        </div>
+                    </tr>
+                </tbody>
+            </table>
             <h2 class="changes-title">Credentials</h2>
             <table>
                 <tbody>
-                <tr>
-                    <td class="label" width="110">User :</td>
-                    <td class="value"
-                        width="80">{{detailsData?.getCredentials().getUser()}}</td>
-                </tr>
-                <tr>
-                    <td class="label" width="110">Password :</td>
-                    <td class="value"
-                        width="80">{{detailsData?.getCredentials().getPassword()}}</td>
-                </tr>
+                    <tr>
+                        <td class="label" width="110">User :</td>
+                        <td class="value" width="80">{{detailsData?.getCredentials().getUser()}}</td>
+                        <td class="label" width="110">Password :</td>
+                        <td class="value" width="80">{{detailsData?.getCredentials().getPassword()}}</td>
```

Comment:
actually - we should probably remove the password <td> altogether. We should never show the true value anyway because it's secret - all we would end up displaying is `******` which is of no use